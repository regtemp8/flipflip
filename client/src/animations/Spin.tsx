import React, { type PropsWithChildren, useState } from 'react'
import { Spring, animated } from 'react-spring/renderprops'

export interface SpinProps {
  className?: string
  title?: string
  style?: any
  onClick?: () => void
}

export default function Spin(props: PropsWithChildren<SpinProps>) {
  const [toggle, setToggle] = useState(false)

  return (
    <Spring
      from={{ transform: 'rotateY(0deg)' }}
      to={{
        transform: toggle ? 'rotateY(180deg)' : 'rotateY(0deg)'
      }}
    >
      {(animation) => (
        <animated.div
          className={props.className}
          style={props.style ? { ...animation, ...props.style } : animation}
          title={props.title}
          onMouseEnter={() => {
            setToggle(!toggle)
          }}
          onClick={props.onClick}
        >
          <Spring
            from={{ transform: 'rotateY(0deg)' }}
            to={{
              transform: toggle ? 'rotateY(-180deg)' : 'rotateY(0deg)'
            }}
          >
            {(animation) => (
              <animated.div style={animation}>{props.children}</animated.div>
            )}
          </Spring>
        </animated.div>
      )}
    </Spring>
  )
}

;(Spin as any).displayName = 'Spin'
