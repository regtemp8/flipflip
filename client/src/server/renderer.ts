import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
