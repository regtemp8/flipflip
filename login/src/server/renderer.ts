import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
