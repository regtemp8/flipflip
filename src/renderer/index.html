<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
