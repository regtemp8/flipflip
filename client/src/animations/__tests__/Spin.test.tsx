import React from 'react'
import { describe, it, expect } from '@jest/globals'
import renderer from 'react-test-renderer'
import Spin from '../Spin'

describe('Spin', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Spin />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
