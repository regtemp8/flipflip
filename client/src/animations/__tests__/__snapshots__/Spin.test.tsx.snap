// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin should match snapshot 1`] = `
<div
  onMouseEnter={[Function]}
  style={
    Object {
      "transform": "rotateY(0deg)",
    }
  }
>
  <div
    style={
      Object {
        "transform": "rotateY(0deg)",
      }
    }
  />
</div>
`;
